backend/src/modules/auth/auth.guard.ts

```diff
@@ -0,0 +1,26 @@
+import {
+    CanActivate,
+    ExecutionContext,
+    Injectable,
+    UnauthorizedException,
+} from '@nestjs/common';
+import { Request } from 'express';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+    async canActivate(context: ExecutionContext): Promise<boolean> {
+        const request = context.switchToHttp().getRequest();
+        const token = this.extractTokenFromHeader(request);
+
+        if (!token || token != process.env.API_TOKEN) {
```

Comment:
I just hardcoded an API_TOKEN as the bearer token, hopefully this is security enough.

Reply:
Reference: https://matthewdavis.io/adding-context-to-your-requests-with-nest-js/

Reply:
@ianliuwk1019 I think your suggestions are more complicated than necessary for this use case. The proxy service should never be called by anything other than FAM API. As long as the API key is there, it should just do anything it is asked. The security is in the logic of the FAM api. If we were making a general service to be used by multiple clients, the security requirements would be much more rigorous. 

I think of this proxy service as just a FAM API function that happens to live over in OpenShift. We should never share it with another team or even another application.

Reply:
After our discussion, changed bearer token to be x-api-key, and will not pass user id token as bearer token for now, all the requester information will still be passed as parameters to the api call.